File: webapi/Classes/EventSeat.cs
using System;
using System.ComponentModel.DataAnnotations;
using webapi.Classes;

namespace webapi.Classes
{
    public class EventSeat
    {
        [Key]
        [Required]
        public Guid Id { get; set; }
        public Guid SeatId { get; set; }
        public Seat Seat { get; set; }
        public Guid EventId { get; set; }
        public Event Event { get; set; }
        public double Price { get; set; }
        public EventSeatStatus Status { get; set; }
    }
}

File: angularapp/src/interfaces/event-seat.ts
import {Seat} from "./seat";
import {Event} from "./event";

export interface EventSeat {
  id: string;
  seatId: string;
  eventId: string;
  price: number;
  event: Event;
  seat: Seat;
  status: EventSeatStatus;
}

export enum EventSeatStatus {
  Available = 0,
  Reserved = 1,
  Sold = 2
}
